service/history/replication/raw_task_converter.go

```diff
@@ -52,77 +52,44 @@ import (
 
 type (
 	SourceTaskConverterImpl struct {
-		historyEngine           shard.Engine
-		namespaceCache          namespace.Registry
-		clientClusterShardCount int32
-		clientClusterName       string
-		clientShardKey          ClusterShardKey
+		historyEngine  shard.Engine
+		namespaceCache namespace.Registry
 	}
 	SourceTaskConverter interface {
 		Convert(task tasks.Task) (*replicationspb.ReplicationTask, error)
 	}
 	SourceTaskConverterProvider func(
 		historyEngine shard.Engine,
 		shardContext shard.Context,
-		clientClusterShardCount int32,
-		clientClusterName string,
-		clientShardKey ClusterShardKey,
+		clientClusterName string, // Task converter is cluster aware.
 	) SourceTaskConverter
 )
 
 func NewSourceTaskConverter(
 	historyEngine shard.Engine,
 	namespaceCache namespace.Registry,
-	clientClusterShardCount int32,
-	clientClusterName string,
-	clientShardKey ClusterShardKey,
 ) *SourceTaskConverterImpl {
 	return &SourceTaskConverterImpl{
-		historyEngine:           historyEngine,
-		namespaceCache:          namespaceCache,
-		clientClusterShardCount: clientClusterShardCount,
-		clientClusterName:       clientClusterName,
-		clientShardKey:          clientShardKey,
+		historyEngine:  historyEngine,
+		namespaceCache: namespaceCache,
 	}
 }
 
 func (c *SourceTaskConverterImpl) Convert(
 	task tasks.Task,
 ) (*replicationspb.ReplicationTask, error) {
-	var shouldProcessTask bool
+
+	var ctx context.Context
+	var cancel context.CancelFunc
 	namespaceEntry, err := c.namespaceCache.GetNamespaceByID(
 		namespace.ID(task.GetNamespaceID()),
 	)
 	if err != nil {
 		// if there is error, then blindly send the task, better safe than sorry
-		shouldProcessTask = true
-	}
-
-	if namespaceEntry != nil {
-	FilterLoop:
-		for _, targetCluster := range namespaceEntry.ClusterNames() {
-			if c.clientClusterName == targetCluster {
-				shouldProcessTask = true
-				break FilterLoop
-			}
-		}
-	}
-
-	if !shouldProcessTask {
-		return nil, nil
-	}
-
-	clientShardID := common.WorkflowIDToHistoryShard(task.GetNamespaceID(), task.GetWorkflowID(), c.clientClusterShardCount)
-	if clientShardID != c.clientShardKey.ShardID {
-		return nil, nil
+		ctx, cancel = context.WithTimeout(context.Background(), applyReplicationTimeout)
```

Comment:
mark the context as lowest priority?

Reply:
actually, just use empty string as nsName and fall through to `newTaskContext` below?

`if there is error, then blindly send the task, better safe than sorry`
^ comment can also be removed I think.